drone/src/agent/backend.rs

```diff
@@ -99,19 +99,16 @@ impl BackendMonitor {
         backend_id: &BackendId,
         engine: &E,
         nc: &TypedNats,
-        mut meta_log_rx: Receiver<DroneLogMessage>,
+        meta_log_rx: ReceiverStream<DroneLogMessage>,
     ) -> JoinHandle<()> {
-        let mut stream = engine.log_stream(backend_id);
+        let mut stream = engine.log_stream(backend_id).merge(meta_log_rx);
         let nc = nc.clone();
         let backend_id = backend_id.clone();
 
         tokio::spawn(async move {
             tracing::info!(%backend_id, "Log recording loop started.");
 
-            while let Some(v) = tokio::select! {
-                v = stream.next() => { v }
-                v = meta_log_rx.recv() => { v }
-            } {
+            while let Some(v) = stream.next().await {
```

Comment:
oh, this is much nicer!

Although, since this is an `mpsc` sender, have you considered cloning the sender? That way instead of merging streams all the messages (from both clones of the sender) could go to the same stream.

Reply:
uhhhh, I don't think I follow, I'm merging the receiver stream with the nats stream so that it only terminates when they're both done.